### PR TITLE
feat(logging): add security logging support

### PR DESCRIPTION
**Pull request by JacopoDaeli**

### CODE REVIEW

**Review by decompil3d — APPROVED** [review]

File: lib/routes/cdn.js [hunk 1]
```diff
@@ -68,6 +68,7 @@ module.exports =
         }
       },
       preHandler: fastify.auth([fastify.verifyAuthentication]),
```
Comment (msluther) [inline lib/routes/cdn.js:70]: `verifyAuthentication` this mechanism is also going to need to have security logging added to it in some way.

File: lib/routes/cdn.js [hunk 2]
```diff
@@ -99,6 +100,19 @@ module.exports =
           const wrhsAssets = await fastify.tarballToAssets(req.raw);
           const { files, metadata: metas } = wrhsAssets;
           await fastify.uploadAssetFilesToStorage(files, expTimestamp);
+
+          if (fastify.log.security) {
+            fastify.log.security({
```
Comment (msluther) [inline lib/routes/cdn.js:105]: this works, but you could also just do this as events that the internal wrappers can listen to

File: lib/routes/cdn.js [hunk 3]
```diff
@@ -99,6 +100,19 @@ module.exports =
...
+              success: true,
+              message: 'Files sucessfully uploaded',
+              method: req.method,
+              url: req.url,
```
Comment (msluther) [inline lib/routes/cdn.js:109]: Will the method + url have enough information to know what changed? is there additional data we might need? do we want to know what files were uploaded here?

File: lib/routes/objects.js [hunk 4]
```diff
@@ -230,6 +230,18 @@ module.exports =
             expiration
           });
 
+          if (fastify.log.security) {
+            fastify.log.security({
+              success: true,
+              message: 'Object variant sucessfully created',
+              method: req.method,
+              url: req.url,
```
Comment (msluther) [inline lib/routes/objects.js:238]: Similarly here, is method + url enough to know what was created?

File: lib/routes/cdn.js (hunk 1 above)
Comment (JacopoDaeli) [reply lib/routes/cdn.js:70]: Yes indeed. This is what we handle in our private repo indeed.

File: lib/routes/cdn.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (JacopoDaeli) [reply lib/routes/cdn.js:105]: I agree. I am working on a plan to create a pino security log function plugin that will replace this. For now this is just an easy way to meet our internal requirements.

File: lib/routes/cdn.js (hunk 3 above)
Comment (JacopoDaeli) [reply lib/routes/cdn.js:111]: I've increased details at message level.

File: lib/routes/objects.js (hunk 4 above)
Comment (JacopoDaeli) [reply lib/routes/objects.js:240]: Same as above.

File: lib/routes/cdn.js (hunk 2 above)
Comment (JacopoDaeli) [reply lib/routes/cdn.js:105]: Certainly a way I'll consider in a fast follow refactoring.

**Pull request merged**